pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -348,13 +349,93 @@ func (p *staticPolicy) GetTopologyHints(s state.State, pod *v1.Pod, container *v
 
 	// Generate hints.
 	cpuHints := p.generateCPUTopologyHints(available, reusable, requested)
-	klog.Infof("[cpumanager] TopologyHints generated for pod '%v', container '%v': %v", pod.Name, container.Name, cpuHints)
+	klog.Infof("[cpumanager] TopologyHints generated for pod '%v', container '%v': %v", format.Pod(pod), container.Name, cpuHints)
 
 	return map[string][]topologymanager.TopologyHint{
 		string(v1.ResourceCPU): cpuHints,
 	}
 }
 
+func (p *staticPolicy) getPodRequestedCPU(pod *v1.Pod) int {
+	// The maximum of requested CPUs by init containers.
+	requestedByInitContainers := 0
+	for _, container := range pod.Spec.InitContainers {
+		if _, ok := container.Resources.Requests[v1.ResourceCPU]; !ok {
+			continue
+		}
+		requestedCPU := p.guaranteedCPUs(pod, &container)
+		if requestedCPU > requestedByInitContainers {
+			requestedByInitContainers = requestedCPU
+		}
+	}
+	// The sum of requested CPUs by app containers.
+	requestedByAppContainers := 0
+	for _, container := range pod.Spec.Containers {
+		if _, ok := container.Resources.Requests[v1.ResourceCPU]; !ok {
+			continue
+		}
+		requestedByAppContainers += p.guaranteedCPUs(pod, &container)
+	}
+
+	if requestedByInitContainers > requestedByAppContainers {
+		return requestedByInitContainers
+	}
+	return requestedByAppContainers
+}
+
+func (p *staticPolicy) GetPodTopologyHints(s state.State, pod *v1.Pod) map[string][]topologymanager.TopologyHint {
+	// Get a count of how many guaranteed CPUs have been requested by Pod.
+	requestedByPod := p.getPodRequestedCPU(pod)
+
+	// Number of required CPUs is not integer or a pod hasn't got a Guaranteed QoS class.
+	// It will be treated by Topology Manager as having no preference and cause to ignore this
+	// resource while considering a pod alignment.
+	// In definition of hints this is equal to: TopologyHints[NUMANodeAffinity: nil, Preferred: true].
```

Comment:
Can we make sure to have the same comment across this function and `GetTopologyHints()` for consistency?
I'm fine using either one, but I think they should be the same.

If you want to use this one, I'd change the wording slightly:
> // Number of required CPUs is not an integer or a pod is not part of the Guaranteed QoS class.
> // It will be treated by the TopologyManager as having no preference and cause it to ignore this
> // resource when considering pod alignment.
> // In terms of hints, this is equal to: TopologyHints[NUMANodeAffinity: nil, Preferred: true].

Reply:
It doesn't look like this was done.

Reply:
Checked and ensured it is done.